components/renku_data_services/errors/errors.py

```diff
@@ -45,6 +45,13 @@ class UnauthorizedError(BaseError):
     quiet: bool = True
 
 
+@dataclass
+class InvalidOAuth2Token(UnauthorizedError):
+    """Raised when a token for a connected service is invalid."""
+
+    message: str = "OAuth2 token for connected service invalid or expired."
```

Comment:
Non-blocking: does it make sense to add a subclass here when this seems to be used only once?